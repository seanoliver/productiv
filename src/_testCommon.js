const TEST_TODOS = [{
	id: 'im-an-id',
	title: 'Test title',
	description: 'Test description',
	priority: 1,
},
{
	id: 'im-also-an-id',
	title: 'Test title 2',
	description: 'Test description 2',
	priority: 1,
},
{
	id: 'me-too',
	title: 'Test title 3',
	description: 'Test description 3',
	priority: 1,
}];

export { TEST_TODOS }