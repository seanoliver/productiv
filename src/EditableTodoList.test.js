import { render, fireEvent } from '@testing-library/react';
import EditableTodoList from './EditableTodoList.js';
import { TEST_TODOS } from './_testCommon.js';

describe('EditableTodoList', function () {
	it('renders without crashing', function () {
		render(
			<EditableTodoList
				todos={TEST_TODOS}
				update={jest.fn()}
				remove={jest.fn()}
			/>
		);
	});

	it('matches snapshot in non-editing state', function () {
		const { container, debug } = render(
			<EditableTodoList
				todos={TEST_TODOS}
				update={jest.fn()}
				remove={jest.fn()}
			/>
		);
		expect(container).toMatchSnapshot();
	});

  it('renders the correct number of EditableTodo components', function () {
		const { container, debug } = render(
			<EditableTodoList
				todos={TEST_TODOS}
				update={jest.fn()}
				remove={jest.fn()}
			/>
		);
    expect(container.querySelectorAll('.EditableTodo').length).toEqual(3);
    });

	});