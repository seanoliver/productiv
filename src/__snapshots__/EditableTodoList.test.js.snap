// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableTodoList matches snapshot in non-editing state 1`] = `
<div>
  <div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b
              class="Todo-title"
            >
              Test title
            </b>
            <small
              class="Todo-priority"
            >
              (priority: 
              1
              )
            </small>
          </div>
          <div>
            <small
              class="Todo-description"
            >
              Test description
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b
              class="Todo-title"
            >
              Test title 2
            </b>
            <small
              class="Todo-priority"
            >
              (priority: 
              2
              )
            </small>
          </div>
          <div>
            <small
              class="Todo-description"
            >
              Test description 2
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b
              class="Todo-title"
            >
              Test title 3
            </b>
            <small
              class="Todo-priority"
            >
              (priority: 
              3
              )
            </small>
          </div>
          <div>
            <small
              class="Todo-description"
            >
              Test description 3
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
